3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -12,22 +12,24 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 
 function rollDiceUntil(wantedValue) {
   // TODO: rewrite this function using async/await
-  const recurse = () => {
-    return rollDice().then((value) => {
-      if (value !== wantedValue) {
-        return recurse();
-      }
-      return value;
-    });
+  const recurse = async () => {
```

Comment:
This code still uses the "recursion" technique, that is _necessary_ if you can't use the async/await syntax, but _isn't necessary anymore_ with the async/await syntax. Here is what you could write it as, without recursion:

```js
async function rollDiceUntil(wantedValue) {
  let value;
  while (value !== wantedValue) {
    value = await rollDice();
  }
  return value;
}
```

The "cool thing" about async/await syntax is that you write your code "as if it wasn't asynchronous in the first place". So, when you want to write something with async/await, it can be helpful to try to think what the code would look like if it was synchronous, in this case:

```js
function rollDiceUntil(wantedValue) {
  let value;
  while (value !== wantedValue) {
    value = rollDice();
  }
  return value;
}
```